### PR TITLE
ci: Increase timeouts for slow builds

### LINKED ISSUE #26818: Coverage/arm64 build time out on a full build
changing the `VERSION.txt` seems to trigger a full build which pretty reliably times out coverage

https://dev.azure.com/cncf/envoy/_build/results?buildId=134429&view=logs&s=4f7d954b-a765-565f-91a2-c04870dab43f&j=bbe4b42d-86e6-5e9c-8a0b-fea01d818a24

we can probably increase the timeout but would be good to optimize somehow

### PR DESCRIPTION
**Pull request by phlax**
Fix #26818 

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: .azure-pipelines/stages.yml [hunk 1]
```diff
@@ -567,7 +567,7 @@ stages:
     dependsOn: []
     # Skip checks if only mobile/ or docs/ have changed.
     condition: and(not(canceled()), or(succeeded(), eq(variables['PostSubmit'], true)), ne(stageDependencies.env.repo.outputs['changed.mobileOnly'], 'true'), ne(stageDependencies.env.repo.outputs['changed.docsOnly'], 'true'), ne(stageDependencies.env.repo.outputs['changed.examplesOnly'], 'true'))
-    timeoutInMinutes: 180
+    timeoutInMinutes: 240
```
Comment (htuch) [inline .azure-pipelines/stages.yml:570]: Just for my education, when do we get into a situation where we have a 4H coverage build that isn't broken?

**Comment by phlax** [comment]
here:

https://dev.azure.com/cncf/envoy/_build/results?buildId=134521&view=logs&j=bbe4b42d-86e6-5e9c-8a0b-fea01d818a24&t=00ade0d6-ffae-5af0-b5e0-e33b1813c8bc&l=4769


for example. This test was run in part to get these timings - its incredibly frustrating to wait 3 hours for a test and it then timeout.

Whether coverage _should_ take 3+ hours is another matter, that i think we should address

**Comment by phlax** [comment]
heres another running the first Pr on the new 1.26 branch (ie full rebuild)

https://dev.azure.com/cncf/envoy/_build/results?buildId=134604&view=logs&s=b2710bd9-3b3a-5a17-3102-13ba462a848c

both arm and cov die from timeout

**Comment by phlax** [comment]
here is a non-scheduled job timing out https://dev.azure.com/cncf/envoy/_build/results?buildId=134635&view=logs&j=bbe4b42d-86e6-5e9c-8a0b-fea01d818a24

i really think this should be landed asap @htuch

**Comment by phlax** [comment]
relatedly im very much hoping that #26839 will begin to address the underlying problem

**Review by htuch — APPROVED** [review]

**Pull request merged**